### PR TITLE
cmd: use multiple withdrawal addresses

### PR DESCRIPTION
**Pull request by dB2510**
Uses multiple withdrawal addresses while creating deposit data.

category: feature
ticket: #1650

### CODE REVIEW

File: eth2util/deposit/deposit.go [hunk 1]
```diff
@@ -67,19 +67,24 @@ func getMessageRoot(pubkey eth2p0.BLSPubKey, withdrawalAddr string) (eth2p0.Root
 }
 
 // MarshalDepositData serializes a list of deposit data into a single file.
-func MarshalDepositData(msgSigs map[eth2p0.BLSPubKey]eth2p0.BLSSignature, withdrawalAddr, network string) ([]byte, error) {
-	creds, err := withdrawalCredsFromAddr(withdrawalAddr)
-	if err != nil {
-		return nil, err
-	}
-
+func MarshalDepositData(msgSigs map[eth2p0.BLSPubKey]eth2p0.BLSSignature, withdrawalAddrByPubkey map[eth2p0.BLSPubKey]string, network string) ([]byte, error) {
```
Comment (dB2510) [inline eth2util/deposit/deposit.go:70]: should we just pass slices of pubkeys, signatures and withdrawal addresses? rather than passing 2 maps

Comment (corverroos) [reply eth2util/deposit/deposit.go:70]: yeah

Comment (dB2510) [reply eth2util/deposit/deposit.go]: done

File: cmd/createcluster.go [hunk 2]
```diff
@@ -226,39 +225,43 @@ func runCreateCluster(ctx context.Context, w io.Writer, conf clusterConfig) erro
...
-		return nil, err
-	}
+// signDepositDatas returns Distributed Validator pubkeys and deposit data signatures corresponding to each pubkey.
+func signDepositDatas(secrets []*bls_sig.SecretKey, withdrawalAddresses []string, network string) ([]eth2p0.BLSPubKey, []eth2p0.BLSSignature, error) {
+	var (
+		pubkeys    []eth2p0.BLSPubKey
+		signatures []eth2p0.BLSSignature
+	)
+	for i, secret := range secrets {
+		withdrawalAddr, err := checksumAddr(withdrawalAddresses[i])
```
Comment (corverroos) [inline cmd/createcluster.go:235]: avoid panic, verify length of addresses is equaled to length of secrets.

File: dkg/dkg.go [hunk 3]
```diff
@@ -539,12 +558,11 @@ func signLockHash(shareIdx int, shares []share, hash []byte) (core.ParSignedData
 }
 
 // signDepositData returns a partially signed dataset containing signatures of the deposit data signing root.
-func signDepositData(shares []share, shareIdx int, withdrawalAddr string, network string) (core.ParSignedDataSet, map[core.PubKey][]byte, error) {
-	withdrawalHex := checksumAddr(withdrawalAddr)
-
+func signDepositData(shares []share, shareIdx int, withdrawalAddresses []string, network string) (core.ParSignedDataSet, map[core.PubKey][]byte, error) {
```
Comment (corverroos) [inline dkg/dkg.go:561]: verify withdrawalAddresses length

File: eth2util/deposit/deposit.go [hunk 4]
```diff
@@ -67,27 +67,27 @@ func getMessageRoot(pubkey eth2p0.BLSPubKey, withdrawalAddr string) (eth2p0.Root
 }
 
 // MarshalDepositData serializes a list of deposit data into a single file.
-func MarshalDepositData(msgSigs map[eth2p0.BLSPubKey]eth2p0.BLSSignature, withdrawalAddr, network string) ([]byte, error) {
-	creds, err := withdrawalCredsFromAddr(withdrawalAddr)
-	if err != nil {
-		return nil, err
-	}
-
+func MarshalDepositData(pubkeys []eth2p0.BLSPubKey, depositDataSigs []eth2p0.BLSSignature, withdrawalAddrs []string, network string) ([]byte, error) {
```
Comment (corverroos) [inline eth2util/deposit/deposit.go:70]: verify withdrawalAddresses length

**Review by corverroos — APPROVED** [review]

File: dkg/dkg.go (hunk 3 above)
Comment (dB2510) [reply dkg/dkg.go:553]: done

**Pull request merged**